### PR TITLE
Add tests

### PR DESCRIPTION
**Pull request by allisonking**
Also does some refactoring

* Moves Wackabase out of Etsy. Instead, it's instantiated by wack.py and passed around from there.
* Removes references to etsy oauth
* Attempts to add a gh workflow
* Separates out `num_sales` into its own file. I think we can think of files as separate tables in a db, so it makes sense for this to be its own lil file

### CODE REVIEW

File: .github/workflows/pytest.yaml [hunk 1]
```diff
@@ -0,0 +1,22 @@
+name: WacksByWarby
+
+on:
+  pull_request:
```
Comment (jzengg) [inline .github/workflows/pytest.yaml:6]: we should run these on push to master as well

File: wacksbywarby/db.py [hunk 2]
```diff
@@ -2,38 +2,53 @@
...
 
-    @staticmethod
-    def get_last_success():
+    def get_last_num_sales(self):
+        try:
+            with open(self.num_sales_path) as f:
+                num = f.read()
+                return int(num)
+        except FileNotFoundError:
+            return 0
```
Comment (jzengg) [inline wacksbywarby/db.py:42]: how about adding an error log here as well for wackwatch?

File: wacksbywarby/tests/test_wack.py [hunk 3]
```diff
@@ -0,0 +1,109 @@
...
+from wacksbywarby.wack import main
+from wacksbywarby.db import Wackabase
+
+
+@pytest.mark.parametrize(
+    "changes, num_additional_sales, num_expected_discord_messages",
+    [
+        # None for num_additional_sales means we'll calculate a proper number
+        ([], None, 0),
+        ([("957749348", -1)], None, 1),
```
Comment (jzengg) [inline wacksbywarby/tests/test_wack.py]: can you comment with what case these are as well? aka 1 sale, increase in inventory (not handled yet so no messages expected)

File: wacksbywarby/wack.py [hunk 4]
```diff
@@ -71,38 +75,41 @@ def await_pizza_party(discord, num_sales):
...
         logger.info("Dry run: %s", dry)
         etsy = Etsy(debug=dry)
         discord = Discord(debug=dry)
-        id_to_listing_diff = etsy.get_inventory_state_diff()
+
+        previous_inventory = db.get_last_entry()
+        current_inventory = etsy.get_inventory_state()
+        id_to_listing_diff = etsy.get_inventory_state_diff(
+            previous_inventory, current_inventory
+        )
```
Comment (jzengg) [inline wacksbywarby/wack.py:93]: what if werby manually increases the inventory. we would return here but wouldn't update the last_num_sales or the inventory? is it easy to add a test case to test that the DBs are updated in addition to the discord messages?

**Review by jzengg — CHANGES_REQUESTED** [review]
I think we should test the DB writes in the test cases as well! Thanks for the contribution!

File: wacksbywarby/wack.py (hunk 4 above)
Comment (allisonking) [reply wacksbywarby/wack.py:93]: if there's an increase in inventory, the `id_to_listing_diff` will not be falsy, so it shouldn't return here. There is already a test case for this— added a comment for it!

**Review by jzengg — APPROVED** [review]

**Pull request merged**